app/src/main/java/fr/free/nrw/commons/nearby/PlacesRepository.java

```diff
@@ -0,0 +1,25 @@
+package fr.free.nrw.commons.nearby;
+
+import fr.free.nrw.commons.contributions.Contribution;
+import fr.free.nrw.commons.location.LatLng;
+import io.reactivex.Completable;
+import javax.inject.Inject;
+
+public class PlacesRepository {
+
+    private PlacesLocalDataSource localDataSource;
+
+    @Inject
+    public PlacesRepository(PlacesLocalDataSource localDataSource) {
+        this.localDataSource = localDataSource;
+    }
+
+    public Completable save(Place place){
+        return localDataSource.savePlace(place);
+    }
+
+    public Place fetchPlace(LatLng latLng){
```

Comment:
Several Nearby pins (Wikidata items) can have the exact same latitude/longitude, it actually happens rather often.

Wouldn't it work better in such cases if the cache indexed on QID rather than on LatLng?

Reply:
Sure, done in the newer commit https://github.com/commons-app/apps-android-commons/pull/5731/commits/a28117397d1dcb5e6d5b56b970727dde11ff5074